### PR TITLE
Feat/credentialz 3

### PR DESCRIPTION
**Pull request by cm-nokia**
probably will need a rebase depending on if/when ondatra pin gets bumped in main

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

### CODE REVIEW

**Review by NishaSadhasivam-work — CHANGES_REQUESTED** [review]

File: feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+	actualHostCertificateVersion := sshServer.GetActiveHostCertificateVersion()
+
+	if actualHostCertificateVersion != hostCertificateVersion {
+		t.Fatalf(
+			"telemetry reports host certificate version is not correct\n\tgot: %s\n\twant: %s",
+			actualHostCertificateVersion, hostCertificateVersion,
+		)
+	}
+
+	actualHostCertificateCreatedOn := sshServer.GetActiveHostCertificateCreatedOn()
```
[the remark this answers is not in the document]
Comment (cm-nokia) [reply feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go:240]: this and the other one are both valid methods in latest ondatra generated structs so im not sure what you want me to do here

File: feature/security/gnsi/credentialz/tests/host_certificates/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Credentialz-3: Host Certificates
+
+## Summary
+
+Test that Credentialz can properly fetch and push SSH host certificates, and that the DUT sends 
+this certificate during SSH authentication.
+
```
[the remark this answers is not in the document]
Comment (cm-nokia) [reply feature/security/gnsi/credentialz/tests/host_certificates/README.md:7]: not sure what the ask is here? this is just straight from the google written test cases [here](https://github.com/openconfig/featureprofiles/tree/main/feature/security/gnsi/credentialz/tests)

File: feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+	err := caCmd.Run()
+	if err != nil {
+		t.Fatalf("failed generating ca key pair, error: %s", err)
+	}
+
+	dutCmd := exec.Command(
+		"ssh-keygen",
+		"-t", "ed25519",
+		"-f", "ca",
```
Comment (sourcequench) [inline feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go:66]: Why is it the dutCmd has -f "ca" and the caCmd has -f "dut"? Seems backwards.

File: feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+	if err != nil {
+		t.Fatalf("failed reading host private key, error: %s", err)
+	}
+
+	rotateHostParameters(
+		t,
+		dut,
+		[]*credentialz.ServerKeysRequest_AuthenticationArtifacts{
+			{
+				PrivateKey: dutPrivateKeyContents,
```
Comment (sourcequench) [inline feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go:131]: Why are we sending along private key only? If you are preparing DUT with an externally created keypair you'd need, well, the public key too.

Comment (dipchauh) [reply feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go:131]: It is expected that the DUT derive public key from the private key itself. Please refer to this gnsi [PR](https://github.com/openconfig/gnsi/pull/63).

File: feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go (hunk 3 above)
Comment (dipchauh) [reply feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go]: Fixed

**Comment by dipchauh** [comment]
Closing this PR due to CLA issues. New PR opened https://github.com/openconfig/featureprofiles/pull/3357.

**Pull request closed without merging**